src/ui/zabapgit_js_common.w3mi.data.js

```diff
@@ -2022,7 +2022,19 @@ function enumerateToolbarActions() {
       if (item.nodeName !== "LI") continue; // unexpected node
       if (item.children.length >=2 && item.children[1].nodeName === "UL") {
         // submenu detected
-        processUL(item.children[1], item.children[0].innerText);
+        var menutext = item.children[0].innerText;
+        // special treatment for menus without text
+        if (!menutext) {

```

Comment:
Good idea, but I'd say a bit indirect implementation. What if icons change ? There is no direct relation between icon and specific menu. Icon may repeat somewhere, etc ... I'd suggest to look into other approaches as well. E.g. `data-palette-prefix` or `data-menu-path` attr - it would be more unambiguos

![image](https://user-images.githubusercontent.com/15635498/95630966-797e8a00-0a8b-11eb-9c2a-fa02cb5e7bf6.png)

![image](https://user-images.githubusercontent.com/15635498/95631170-e2fe9880-0a8b-11eb-93e1-8a7863bca6a6.png)